src/rc.rs

```diff
@@ -542,9 +535,6 @@ impl<T: ?Sized, const NUM: usize, const DEN: usize> StaticRc<T, NUM, DEN> {
     where
         AssertEqType!(NUM, A + B): Sized,
     {
-        #[cfg(debug_assertions)]
-        let (left, right) = Self::validate_pair(left, right);
```

Comment:
What is your motivation for removing those debug aids?

Reply:
I found it surprising that an "unchecked" function was still preforming checks. I had assumed joining would simply just forget one side and transmute the other, which is actually what I wanted for my use case. In my opinion there should be supported ways to opt out of the validity requirement for this crate. Given that it generally doesn't seem to be unsound nor UB. Leaking memory is safe.

Reply:
>  In my opinion there should be supported ways to opt out of the validity requirement for this crate.

I see.

I agree that opting out -- temporarily -- of the validity requirement for this crate should definitely be possible. An unchecked `into_parts` + `from_parts`, and an unchecked `transmute`, sound like worthy additions in this regard.

> Given that it generally doesn't seem to be unsound nor UB. Leaking memory is safe.

You are correct that leaking memory is safe, but that is not all that is happening here. If you were to accidentally pass two `StaticRc` with mismatching pointers:

 - You would be forgetting part of the ownership of the `right` pointer -- which would ultimately lead to a memory leak. That is _sound_.
 - You would, however, be "claiming" more ownership of the `left` pointer than you ought to -- which would ultimately lead to releasing its memory whilst other `StaticRc` to it, aka use-after-free, which is definitely _unsound_.

This is why it is actually unsound to have mismatching pointers in `join_unchecked`, and why the method is labelled `unsafe`.

The method, in the tradition of the standard library, is not marked "unchecked" as a guarantee that no check is ever performed, but instead as a note that not all checks may be performed and it's thus the responsibility of the caller to pay attention to the safety invariants. It then actually performs the checks in Debug for "friendliness" reasons: notifying the user of mistakes, and elides them in Release for performance reason.